plugins/GSdx/GSDevice.cpp

```diff
@@ -133,9 +133,9 @@ void GSDevice::Present(GSTexture* sTex, GSTexture* dTex, const GSVector4& dRect,
 
 GSTexture* GSDevice::FetchSurface(int type, int w, int h, bool msaa, int format)
 {
-	GSVector2i size(w, h);
+	const GSVector2i size(w, h);
 
-	for(list<GSTexture*>::iterator i = m_pool.begin(); i != m_pool.end(); ++i)
+	for(std::deque<GSTexture*>::const_iterator i = m_pool.cbegin(); i != m_pool.cend(); ++i)
```

Comment:
Honestly it does't need to be clever. You don't need thousand of textures. 300 (unallocated) textures is already quite huge.